### DISCUSSION TITLE
Configuring Port Forwarding with PIA

### DISCUSSION THREAD

**Discussion by metaclam** [haugene/docker-transmission-openvpn#2840] [Q&A] [answered]
anyone know how to do this with this container or better yet can share a working configuration?

I've learned that some private trackers require an open port outside of some common BitTorrent ranges. 

I've got no idea how to do this on any VPN let alone PIA Which apparently changes the one available port.

**Comment by metaclam** [comment]
Surely this is something many people need to do with torrenting?   Anyone can help?  

I was referred to [this by PIA support](https://helpdesk.privateinternetaccess.com/guides/linux/alternative-setups-4/linux-manual-connection-scripts
), but I presume some of this is already implemented in the container.  So how to configure port forwarding?

**Comment by metaclam** [comment]
Hmm…  it seems like this port forwarding should be built-in to the container after all?

Referring to [this discussion](https://github.com/haugene/docker-transmission-openvpn/discussions/2668), I stopped the container and added the missing (or unset) keys listed by jcxpgh78 in his config file, including:  

UFW_EXTRA_PORTS=48595   {<--- or actual peer port #}
DISABLE_PORT_UPDATER=false
UFW_DISABLE_IPTABLES_REJECT=false

Did above in Container Manager settings for the container.

In the Container Manager "ports settings" I mapped 48595 local port to 48595 container port.

On my router I forwarded port 48595 (example) to the Synology IP (reserved DHCP)

This all seems to work as is -- (great) -- except that the ["peer-port" setting in settings.json keeps getting reset to 1,](https://forum.transmissionbt.com/viewtopic.php?t=21250) depending on what WebUI I am using. This is happening with both Transmission Remote GUI (on Mac) and an Shift, at least.

**Reply by pkishino** [reply]
No, don’t do that.. the port forwarding should be inside the vpn connection..Pia already has a script that sets the peering port to an open one.

**Reply by metaclam** [reply]
i'm sorry. what's "that" that I should not do?  I did several things.

**Reply by bassmadrigal** [reply, marked as answer]
Port forwarding with PIA is automatically enabled within the container and does not need configuration within the docker settings. The only thing I've done in my docker compose conf file is add this line `- OPENVPN_CONFIG=ca_vancouver` to connect to one of the faster PIA server regions that support port forwarding (since none in the US do), but I don't believe that is necessary to get port forwarding.

The port forward test in transgui's settings shows that port forwarding is working fine.

There is no reason to forward the port on your router as the internet doesn't ever see your router, it sees the VPN server. PIA's servers need to forward the port to your connection, which is not touched by your router.

**Reply by metaclam** [reply]
Thank you so much. That was exactly the lucid answer for a newbie I needed. Ironically it was mere hours too late as I finally figured that out myself (with even more googling and a chat with a PIA support person).

I still am unclear whether to enable port forwarding I needed to subscribe to a "dedicated IP" which I did as it seemed the only way.  **Is that correct?**  (If not I suppose it is a small price to pay and may have other uses. If so, that was not at all obvious to me!)

On the container side, you're right, there's no need to specify a port or even that dedicated IP.  The container apparently handles it automatically and sets the peer port within transmission, which yes, as you say, tests OK with Trans Remote GUI!  So all is working finally!

The  process of determining which servers support **both** port forwarding **and** dedicated IP, and then determining what the port actually is (though turns out not needed) was also not straightforward (let alone obvious) to me.  Also it was not obvious how to specify "ca_toronto" if the server was Toronto.  I never found a chart for that, but just (apparently correctly) guessed.

I wrote up a [guide that might have made sense to me](https://github.com/haugene/docker-transmission-openvpn/discussions/2845) from the start.  What a frustrating process

**Reply by wally-crunk** [reply]
> I still am unclear whether to enable port forwarding I needed to subscribe to a "dedicated IP"

Per my testing, no -- PIA can turn provide port forwarding **without** subscribing to a dedicated IP.

**Reply by Presjar** [reply]
Does a method exist to change transmission port to match the provided PIA forwarded port? From what I read the port is assigned random on each connection.

Edit: I see the auto script is already doing this job. So good! Thanks!
